src/reflect/scala/reflect/internal/Trees.scala

```diff
@@ -1552,9 +1552,9 @@ trait Trees extends api.Trees {
   def Block(stats: Tree*): Block = {
     if (stats.isEmpty) Block(Nil, Literal(Constant(())))
     else stats match {
-      case Seq(b @ Block(_, _)) => b
-      case Seq(stat) => Block(stats.toList, Literal(Constant(())))
-      case Seq(_, rest @ _*) => Block(stats.init.toList, stats.last)
+      case collection.Seq(b @ Block(_, _)) => b
+      case collection.Seq(stat) => Block(stats.toList, Literal(Constant(())))
+      case collection.Seq(_, rest @ _*) => Block(stats.init.toList, stats.last)
```

Comment:
Gotcha: Varags are not Seqs anymore. This was the ugliest side-effect that I encountered. The code still compiled but some tests failed.

Reply:
Why can't we let Seq.unapply handle scala.collection.Seqs?

Reply:
This is tricky. Many people have gotten used to the following behavior:

```scala
scala> val xs = scala.collection.mutable.ListBuffer(1, 2, 3)
xs: scala.collection.mutable.ListBuffer[Int] = ListBuffer(1, 2, 3)

scala> xs match { case Seq(xs @ _*) => xs }
res0: Seq[Int] = ListBuffer(1, 2, 3)
```

Technical reason may have been that `scala.Seq` was `scala.collection.Seq` etc, but this source compatibility break is subtle, so I think it's worth being hand-holding about `2.[0 - 12]` code.

Reply:
I'm thinking https://en.wikipedia.org/wiki/Robustness_principle.

Having `scala.Seq` be different to `scala.collection.Seq` is very confusing.  The fact that Scala's pattern matching isn't exhaustive further aggravates the situation.  Could we not introduce `scala.Varargs` instead?